src/components/Header/HeaderCart/HeaderCart.jsx

```diff
@@ -0,0 +1,62 @@
+import useWindowWidth from "../../../../hooks/useWindowResize.js";
+import CartList from "./CartList/CartList.jsx";
+import './HeaderCart.css';
+import CartImg from "../../../../public/img/cart.png";
+
+export default function HeaderCart(props) {
+    const {
+        headerCartProducts,
+        setHeaderCartProducts,
+        setIsOpenHeaderCartModal,
+        isOpenHeaderCartModal,
+        isCartShakingAnimation,
+        setIsCartShakingAnimation,
+    } = props;
+
+    const productQuantity = headerCartProducts.length;
+    let getEachProductSum = headerCartProducts.map(product => product.sum);
+    let getAllProductSum  = getEachProductSum.reduce((sum, current) => sum + current, 0);
+    let width = useWindowWidth();
+
+    function handleOpenCartList() {
```

Comment:
де префікс is? 

Також в тебе цей хендлер не просто відкриває модалку, але й закриває через раз. 

Тому перейменуй на тоггл: 

```suggestion
    function handleToggleIsOpenCartList() {
```

Reply:
Ми памятаєм, як я і говорив раніше запамятати просте правило: Якщо ти обновляєш стейт базуючись на його попередньому значенні, - то тобі треба обновляти його через коллбек. Параметром твій сеттер стейта вже приймає значення стейта. Наприклад:

<img width="592" alt="image" src="https://github.com/viktor-fedoruk/veggy-store-/assets/35539661/805fcd53-8c60-462c-afd3-69e15ae800bf">

Тому просто: setIsOpenHeaderCartModal((isOpenHeaderCartModal) => !isOpenHeaderCartModal);